stripe/src/main/java/com/stripe/android/model/BankAccount.kt

```diff
@@ -95,12 +93,12 @@ data class BankAccount internal constructor(
     val status: Status? = null
 ) : StripeModel {
 
-    @Retention(AnnotationRetention.SOURCE)
-    @StringDef(BankAccountType.COMPANY, BankAccountType.INDIVIDUAL)
-    annotation class BankAccountType {
-        companion object {
-            const val COMPANY: String = "company"
-            const val INDIVIDUAL: String = "individual"
+    enum class Type(internal val code: String) {
+        Company("company"),
+        Individual("individual");
+
+        internal companion object {
+            internal fun fromCode(code: String?) = values().firstOrNull { it.code == code }
```

Comment:
I like this pattern but it would be nice if we could put it in an interface. However, I'm not sure how to make the type constraints work